File: prob-1-multiples-of-3-and-5/prob-1-multiples-of-3-and-5.py
total = 0

'''for i in range(1000):
    if i % 3 == 0 or i % 5 == 0:
        total += i'''
#Optimized Solution

def sumDivisibleBy(n):
    p = int(999 / n)
    return int(n * p * (p + 1) / 2)

total = sumDivisibleBy(3) + sumDivisibleBy(5) - sumDivisibleBy(15)

print(total)

File: prob-1-multiples-of-3-and-5/prob-1-multiples-of-3-and-5.go
package main

import . "fmt"

func sumDivisibleBy(n int) int {
	p := 999 / n
	return int(n * p * (p + 1) / 2)
}

func main() {
	total := 0

	/*for i := 1; i < 1000; i++ {
		if (i%3 == 0) || (i%5 == 0) {
			total += i
		}
	}*/

	//Optimized Solution

	total = sumDivisibleBy(3) + sumDivisibleBy(5) - sumDivisibleBy(15)

	Println(total)
}
